# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContextHelper.h
# Commit: 5ad078db7d7b
# Full Hash: 5ad078db7d7b025ae400922fde028352a430592d
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 21:58:52
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 1. Rename CanvasRenderingContextHelper::GetContext to GetOrCreateContext. r=jgilbert
#   
#   This patch contains no functional changes but sets us up for refactoring
#   in a later part in this series. It also helps silence a bunch of
#   compiler warnings.
# ==============================================================================

diff -r e59ba9934ece -r 5ad078db7d7b dom/canvas/CanvasRenderingContextHelper.h
--- a/dom/canvas/CanvasRenderingContextHelper.h	Fri Dec 10 02:57:51 2021 +0000
+++ b/dom/canvas/CanvasRenderingContextHelper.h	Fri Dec 10 02:57:51 2021 +0000
@@ -39,10 +39,6 @@
  public:
   CanvasRenderingContextHelper();
 
-  virtual already_AddRefed<nsISupports> GetContext(
-      JSContext* aCx, const nsAString& aContextId,
-      JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv);
-
   virtual bool GetOpaqueAttr() = 0;
 
  protected:
@@ -64,6 +60,14 @@
               JS::Handle<JS::Value> aParams, bool aUsePlaceholder,
               ErrorResult& aRv);
 
+  already_AddRefed<nsISupports> GetOrCreateContext(
+      JSContext* aCx, const nsAString& aContextId,
+      JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv);
+
+  already_AddRefed<nsISupports> GetOrCreateContext(
+      JSContext* aCx, CanvasContextType aContextType,
+      JS::Handle<JS::Value> aContextOptions, ErrorResult& aRv);
+
   virtual already_AddRefed<nsICanvasRenderingContextInternal> CreateContext(
       CanvasContextType aContextType);
 